winlogtimeline/collector/parser.py

```diff
@@ -51,115 +51,259 @@ def parser(raw, record):
 
 
 def parse_id_1074(raw, record):
-    record["description"] = "Shutdown Initiated"
+    data = raw["Event"]["EventData"]["Data"]
+
+    record["account"] = data[6]["#text"].split("\\")[-1]
+    if data[3]["#text"] == "0x500ff":
+        record["description"] = "Shutdown Initiated"
+    else:
+        record["description"] = "ERROR"
+    record["details"] = f'Cause: {data[4]["#text"]}'
+    record["computer_name"] = data[1]["#text"]
+
     return record
 
 
 def parse_id_1102(raw, record):
+    data = raw["Event"]["EventData"]["Data"]
+
     record["description"] = "Log Cleared"
+    record["account"] = data[1]["#text"]
+    record["details"] = f'Security event log was cleared by the following account: ' \
+                        f'{data[2]["#text"]}\\{data[1]["#text"]}'
+    record["session_id"] = data[3]["#text"]
+
     return record
 
 
 def parse_id_4616(raw, record):
+    data = raw["Event"]["EventData"]["Data"]
+
     record["description"] = "Time Change"
+    record["account"] = data[1]["#text"]
+    record["details"] = f'Changed To: {data[5]["#text"][0:10]} {data[5]["#text"][11:23]} (UTC) | ' \
+                        f'Previous Time: {data[4]["#text"][0:10]} {data[4]["#text"][11:23]} (UTC)'
+
     return record
 
 
 def parse_id_4624(raw, record):
-    record["description"] = "Logon Events"
+    data = raw["Event"]["EventData"]["Data"]
+
+    record["account"] = data[5]["#text"]
+    reason = data[8]["#text"]
+
+    if reason == "2":
+        record["description"] = "Interactive Logon"
+    elif reason == "3":
+        record["description"] = "Network Connection"
+        try:
+            record["details"] = f'From: {data[11]["#text"]} ({data[18]["#text"]}) using ' \
+                                f'{data[10]["#text"]} auth as {data[6]["#text"]}\\{data[5]["#text"]}'
+        except:
+            record["details"] = f'From: {data[11]["@Name"]} ({data[18]["#text"]}) using ' \
+                                f'{data[10]["#text"]} auth as {data[6]["#text"]}\\{data[5]["#text"]}'
+    elif reason == "7":
+        record["description"] = "Unlock Workstation"
+    elif reason == "10":
+        record["description"] = "Remote Interactive Logon"
+        record["details"] = f'From: {data[18]["#text"]} using {data[10]["#text"]} auth as ' \
+                            f'{data[6]["#text"]}\\{data[5]["#text"]}'
+    elif reason == "11":
+        record["description"] = "Interactive Logon"
+        record["details"] = f'Logon as {data[6]["#text"]}\\{data[5]["#text"]} using cached credentials'
+
+    record["session_id"] = data[7]["#text"]
+
     return record
 
 
 def parse_id_4634(raw, record):
-    record["description"] = "Logoff (Network Connection)"
+    data = raw["Event"]["EventData"]["Data"]
+
+    record["account"] = data[4]["#text"]
+    record["description"] = "Logoff"
+
+    if data[4]["#text"] == "3":
+        record["details"] = "End of Network Connection session"
+    else:
+        record["details"] = "EXCLUDED"
```

Comment:
Checkout out this branch locally and ran it. All log off events show up as 'EXCLUDED'. Is this expected?

Reply:
<img width="602" alt="screen shot 2017-12-01 at 11 27 05 am" src="https://user-images.githubusercontent.com/5000360/33499653-9e4e1c5c-d68a-11e7-8c3b-48342df14a77.png">
I'm only matching what the Excel Macro has.